streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -288,6 +288,19 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                 //  -> don't pass in `task.timeout.ms` to stay responsive if `KafkaStreams#close` gets called
                 final ConsumerRecords<byte[], byte[]> records = globalConsumer.poll(requestTimeoutPlusTaskTimeout);
                 if (records.isEmpty()) {
+                    // get consumer position to step over potential TX markers
+                    offset = retryUntilSuccessOrThrowOnTaskTimeout(
+                        () -> globalConsumer.position(topicPartition),
+                        String.format(
+                            "Failed to get position for partition %s. The broker may be transiently unavailable at the moment.",
+                            topicPartition
+                        )
+                    );
+
+                    if (offset == highWatermark) {
```

Comment:
very nit: maybe `>=`?

Reply:
Could it mask a bug? `offset` should never be larger than `highWatermark` ?

Reply:
Good catch. I also updated the tests to cover this case.